pyincore/analyses/saltlakecge/saltlakecge.py

```diff
@@ -1890,30 +1939,31 @@ def run_solver(cons_filename, temp_file_name="tmp.py"):
         
         '''
 
-
-         ######## The following is for random shocks. ########
+        # The following is for random shocks. #
         sims = sector_shocks
-        iNum = 1 # dynamic model itterations
-        #iNum = len(sims.columns)
+        iNum = 1  # dynamic model itterations
+        # iNum = len(sims.columns)
         KS00 = KS0.copy()
 
+        solver_status = False
         for num in range(iNum):
-            KS0.loc[K,I] = KS00.loc[K,I].mul(sims.iloc[:,num])
+            KS0.loc[K, I] = KS00.loc[K, I].mul(sims.iloc[:, num])
             KS0 = KS0.fillna(0.0)
-            run_solver(filename, tmp)
-
-        domestic_supply, gross_income, household_count, pre_disaster_demand, post_disaster_demand = \
-            gams_to_dataframes(iNum, vars, H, L, soln)
-
-        self.set_result_csv_data("domestic-supply", domestic_supply, name="domestic-supply", source="dataframe",
-                                 index=True)
-        self.set_result_csv_data("pre-disaster-factor-demand", pre_disaster_demand,
-                                 name="pre-disaster-factor-demand", source="dataframe", index=True)
-        self.set_result_csv_data("post-disaster-factor-demand", post_disaster_demand,
-                                 name="post-disaster-factor-demand", source="dataframe", index=True)
-        self.set_result_csv_data("gross-income", gross_income, name="gross-income", source="dataframe", index=True)
-        self.set_result_csv_data("household-count", household_count, name="household-count", source="dataframe",
-                                 index=True)
+            solver_status = run_solver(filename, tmp)
+
+        if solver_status:
+            domestic_supply, gross_income, household_count, pre_disaster_demand, post_disaster_demand = \
+                gams_to_dataframes(iNum, vars, H, L, soln)
+
+            self.set_result_csv_data("domestic-supply", domestic_supply, name="domestic-supply", source="dataframe",
+                                     index=True)
+            self.set_result_csv_data("pre-disaster-factor-demand", pre_disaster_demand,
+                                     name="pre-disaster-factor-demand", source="dataframe", index=True)
+            self.set_result_csv_data("post-disaster-factor-demand", post_disaster_demand,
+                                     name="post-disaster-factor-demand", source="dataframe", index=True)
+            self.set_result_csv_data("gross-income", gross_income, name="gross-income", source="dataframe", index=True)
+            self.set_result_csv_data("household-count", household_count, name="household-count", source="dataframe",
+                                     index=True)
```

Comment:
is it worth it to raise an error here or where you evaluate the terminal condition? This way datawolf step might become "FAILED" and then frontend widget won't be clickable. 
```
else:
    raise ValueError("some message") 
```

Reply:
I will push this